common/models/user.js

```diff
@@ -668,6 +668,45 @@ module.exports = function(User) {
       next();
     });
 
+    // Delete old sessions once email is updated
+    var oldEmail, newEmail, oldId, newId;
+    UserModel.observe('before save', function(ctx, next) {
+      if (ctx.instance) {
+        if (ctx.instance.id !== undefined) {
```

Comment:
If `ctx.instance.id` is `undefined`, then there is no way how to find the affected user model. IIRC this happens only when a new model is going to be created, which means you can bail out via `return next()`.